src/js/utils/error.js

```diff
@@ -0,0 +1,91 @@
+export const jwError = (category, code, severity, data) =>{
+    const message = createMessage(category, code, severity, data);
+    return {
+        category,
+        code,
+        severity,
+        message
+    };
+};
+
+export const errorCategory = {
+    PLAYER: 1,
+    SETUP: 2,
+    MEDIA: 3,
+    NETWORK: 4,
+    AD: 5
+};
+
+export const errorCode = {
+    // An HTTP request returned with a status indicating failure
+    BAD_HTTP_STATUS: 4001,
+    // Crossdomain error
+    CROSSDOMAIN_ERROR: 4002,
+    // Accessing HTTP over HTTPS
+    BLOCKED_MIXED_CONTENT: 4003
+};
+
+export const errorSeverity = {
+    RECOVERABLE: 1,
+    FATAL: 2
+};
```

Comment:
I'd rather have a file `error-category.js` and inside `export const PLAYER = 1` such that you can `import * as ErrorCategory from error-category`. Need to balance developer ease with minification.